conans/test/integration/toolchains/meson/test_mesontoolchain.py

```diff
@@ -165,3 +168,127 @@ def generate(self):
     content = t.load(MesonToolchain.native_filename)
     assert "wrap_mode " not in content
     assert "nofallback" not in content
+
+
+@pytest.mark.skipif(platform.system() != "Windows", reason="requires Win")
+# @pytest.mark.tool("visual_studio", "17")
+# @pytest.mark.tool("clang", "16")
+def test_clang_cl_vscrt_mdd():
```

Comment:
It seems that these tests are very similar, it would be great to factorize (using ``pytest.mark.parametrize``) to reduce code. If not sure how to do it, don't worry, just tell, we can help with that.

Reply:
Great thanks for the pointer, I assumed there'd be a nice way... hopefully I got it right.

What do you think about the VS17 / clang 15 or 16 issue?

Reply:
Yes, looks good, thanks!

I don't think that using the full tool vs/clang or whatever is necessary, the testing you did sounds like the right balance.

Reply:
> In the tests I couldn't get the VS17/clang16 pytest.mark.tool working, so they are commented out. I'm sure I'm just missing some magic incantation somewhere despite VS2022 being enable in my env.

If you tried making it work in CI, only ``functional`` tests have the tools installed, using ``pytest.mark.tool`` will fail.

The tests as you did them don't need the tools installed at all, just Conan, so things are looking good.